sphinx/domains/c/__init__.py

```diff
@@ -745,7 +745,7 @@ def process_doc(self, env: BuildEnvironment, docname: str,
     def process_field_xref(self, pnode: pending_xref) -> None:
         pnode.attributes.update(self.env.ref_context)
 
-    def merge_domaindata(self, docnames: list[str], otherdata: dict[str, Any]) -> None:
+    def merge_domaindata(self, docnames: Set[str], otherdata: dict[str, Any]) -> None:
```

Comment:
```suggestion
    def merge_domaindata(self, docnames: set[str], otherdata: dict[str, Any]) -> None:
```

(might need a `from __future__ import annotations`)

Reply:
`Set` is immutable. `set` is mutable. I don't think we want to encourage mutating these sets, even though it is technically possible?

Reply:
makes sense!